backend/dataall/core/stacks/api/resolvers.py

```diff
@@ -68,50 +53,16 @@ def resolve_task_id(context, source: Stack, **kwargs):
 
 
 def get_stack_logs(context: Context, source, environmentUri: str = None, stackUri: str = None):
-    with context.engine.scoped_session() as session:
-        stack = EnvironmentService.get_stack(session=session, uri=environmentUri, stack_uri=stackUri)
-        if not stack.EcsTaskArn:
-            raise exceptions.AWSResourceNotFound(
-                action='GET_STACK_LOGS',
-                message='Logs could not be found for this stack',
-            )
-
-        query = f"""fields @timestamp, @message, @logStream, @log as @logGroup
-                | sort @timestamp asc
-                | filter @logStream like "{stack.EcsTaskArn.split('/')[-1]}"
-                """
-        envname = os.getenv('envname', 'local')
-        results = CloudWatch.run_query(
-            query=query,
-            log_group_name=f"/{Parameter().get_parameter(env=envname, path='resourcePrefix')}/{envname}/ecs/cdkproxy",
-            days=1,
-        )
-        log.info(f'Running Logs query {query}')
-        return results
+    StackService.get_stack_logs(environmentUri, stackUri)
 
 
 def update_stack(context: Context, source, targetUri: str = None, targetType: str = None):
-    with context.engine.scoped_session() as session:
-        stack = StackRepository.update_stack(session=session, uri=targetUri, target_type=targetType)
-    StackService.deploy_stack(stack.targetUri)
-    return stack
+    return StackService.update_stack_by_target_uri(targetUri, targetType)
```

Comment:
General comment about the refactoring...

* doesn't the existence of the `resolvers.py` become redundant?  They are now just proxy calls so can't we just point the resolver (where the GraphQL query is defined) directly to the service?
* by creating the session within the business logic it's going to be more difficult to unit-test those methods because it will by harder to mock the results from the db. Right?

Reply:
Well, it's open question yet. At least it gives us asyncronous calls (like in get_logs)

Reply:
The principle was that in resolvers we check the api input and in service we just execute business logic. In the PRs for refactoring we are adding validation in the service layer but not in the resolvers. I did not add any comment in these PRs, because we already talked about implementing a better validation mechanism. I opened [an issue ](https://github.com/data-dot-all/dataall/issues/1183)so that we do not forget about it